provider.go

```diff
@@ -5,16 +5,20 @@ package linode
 import (
 	"context"
 	"fmt"
+	"sync"
 
 	"github.com/libdns/libdns"
 	"github.com/linode/linodego"
 )
 
 // Provider facilitates DNS record manipulation with Linode.
 type Provider struct {
-	APIToken string `json:"api_token,omitempty"`
-	Domain   string `json:"domain_id,omitempty"`
-	client   linodego.Client
+	APIToken   string `json:"api_token,omitempty"`
+	APIURL     string `json:"api_url,omitempty"`
+	APIVersion string `json:"api_version,omitempty"`
+	Domain     string `json:"domain_id,omitempty"`
```

Comment:
```suggestion
	DomainID     string `json:"domain_id,omitempty"`
```

I suspect the domain and its ID are very different things, so just for clarity...